Lib/concurrent/futures/process.py

```diff
@@ -520,6 +543,17 @@ def get_n_children_alive(self):
         # This is an upper bound on the number of children alive.
         return sum(p.is_alive() for p in self.processes.values())
 
+    def _spawn_replacement_process(self):
+        p = self.mp_context.Process(
+            target=_process_worker,
+            args=(self.call_queue,
+                  self.result_queue,
+                  self.initializer,
+                  self.initargs,
+                  self.maxtasksperchild))
+        p.start()
+        self.processes[p.pid] = p
```

Comment:
Can you factor this out into `ProcessPoolExecutor`? Ideally, this method can just be:
```python
def _spawn_replacement_process(self):
    executor = self.executor_reference()
    if executor is not None:
        self._adjust_process_count()
```

Reply:
Yes this makes much more sense. I was originally worried that the idle semaphore check in `_adjust_process_count` might mess things up, but after re-examining the code, I don't think this is true.

Reply:
Well, `_add_process_to_pool` and `_spawn_replacement_process` are still duplicated, aren't they?

Reply:
Sorry, I forgot to delete `_spawn_replacement_process`.